divoom/protocol.py

```diff
@@ -202,29 +226,39 @@ def mask(_bytes):
             ret.append(b)
     return ret
 
+
 def unmask(_bytes):
     ret = []
     i = 0
     while i < len(_bytes):
         b = _bytes[i]
         if b == 0x3:
-            b = _bytes[i+1] - 0x3
+            b = _bytes[i + 1] - 0x3
             i += 1
         ret.append(b)
         i += 1
     return ret
 
+
 def freq_to_bytes(freq):
-    freq = freq * 10 # 100.3 => 1003
-                     # 88.1 => 881
+    freq = freq * 10  # 100.3 => 1003
+    # 88.1 => 881
     if freq > 1000:
         ret = [int(freq - 1000), int(freq / 100)]
     else:
         ret = [int(freq % 100), int(freq / 100)]
     return ret
 
+
 def bytes_to_freq(_bytes):
     assert len(_bytes) == 2
     first = _bytes[1] * 10
     second = _bytes[0] / 10
     return first + second
+
+
+class DitooProShowImageCommand(Command):
```

Comment:
Thoughts on this pattern?

Reply:
I like it - looking back to my original code base I think the user-facing API should've looked more like

```python
with Timebox(addr) as dev:
    dev.set_brightness(Brightness.MAX)
    dev.set_radio(98.5)
```

while having a base `Device` class and a base `Command` class

Reply:
Updated and mocked out what that might look like. I'm not sure if I'm up for mapping out all the methods, but I can take an initial stab at it (assuming the interface looks reasonable).

Reply:
Oh, I didn't mean to imply that you should make the changes, was reflecting on what I should've done to start with.

I do like how it turned out, so it is fine with me to merge. I can map out the other commands later

Reply:
Cool, I think this PR should be good to go. Lemme know if you want help/testing on the other commands.